src/backend/distributed/commands/schema.c

```diff
@@ -134,3 +138,67 @@ PlanAlterObjectSchemaStmt(AlterObjectSchemaStmt *alterObjectSchemaStmt,
 
 	return NIL;
 }
+
+
+/*
+ * EnsureSchemaExistsOnAllNodes connects to all nodes with citus extension user
+ * and creates the schema of the given relationId. The function errors out if the
+ * command cannot be executed in any of the worker nodes.
+ */
+void
+EnsureSchemaExistsOnAllNodes(Oid relationId)
+{
+	List *workerNodeList = ActivePrimaryNodeList();
+	ListCell *workerNodeCell = NULL;
+	StringInfo applySchemaCreationDDL = makeStringInfo();
+
+	Oid schemaId = get_rel_namespace(relationId);
+	const char *createSchemaDDL = CreateSchemaDDLCommand(schemaId);
+	uint64 connectionFlag = FORCE_NEW_CONNECTION;
+
+	if (createSchemaDDL == NULL)
+	{
+		return;
+	}
+
+	appendStringInfo(applySchemaCreationDDL, "%s", createSchemaDDL);
+
+	foreach(workerNodeCell, workerNodeList)
+	{
+		WorkerNode *workerNode = (WorkerNode *) lfirst(workerNodeCell);
+		char *nodeName = workerNode->workerName;
+		uint32 nodePort = workerNode->workerPort;
+		MultiConnection *connection =
+			GetNodeUserDatabaseConnection(connectionFlag, nodeName, nodePort,
+										  NULL, NULL);
```

Comment:
We're passing NULL as the user here. That means the current user. To connect as superuser we need to use `CitusExtensionOwner()`.